components/builder-worker/src/runner/mod.rs

```diff
@@ -318,43 +345,36 @@ impl Runner {
         self.teardown();
     }
 
-    pub fn run(mut self, tx: &mpsc::Sender<Job>) -> Result<()> {
+    pub async fn run(mut self, mut tx: async_mpsc::Sender<Job>) -> Result<()> {
         // TBD (SA) - Spin up a LogStreamer first thing indpendendly of setup.
         // Currently we need to do it as part of setup because the log file to be
         // streamed lives inside of the workspace, which is created by setup.
-        let mut streamer = self.do_setup(&tx)?;
+        let mut streamer = self.do_setup(&mut tx).await?;
 
-        self.do_validate(&tx, &mut streamer)?;
-        self.do_install_key(&tx, &mut streamer)?;
-        self.do_clone(&tx, &mut streamer)?;
+        self.do_validate(&mut tx, &mut streamer).await?;
+        self.do_install_key(&mut tx, &mut streamer).await?;
+        self.do_clone(&mut tx, &mut streamer).await?;
 
-        let archive = self.do_build(&tx, &mut streamer)?;
-        self.do_export(&tx, &mut streamer)?;
-        self.do_postprocess(&tx, archive, &mut streamer)?;
+        let archive = self.do_build(&mut tx, &mut streamer).await?;
+        self.do_export(&mut tx, &mut streamer).await?;
+        self.do_postprocess(&mut tx, archive, &mut streamer).await?;
 
         self.cleanup();
         self.complete();
-        tx.send(self.workspace.job).map_err(Error::Mpsc)?;
+        tx.send(self.workspace.job).await.unwrap(); // map_err(Error::Mpsc)?;
```

Comment:
Can we use `map_err(Error::Mpsc)?;` here?

Reply:
Addressed in a33367f0febdd930dfb5cba3892af203fc5b3a51